src/SqsPoller/SqsPollerHostedService.cs

```diff
@@ -64,35 +65,48 @@ private async Task Handle(string queueUrl, SemaphoreSlim semaphore, Cancellation
                     "Start processing the message with id {message_id} and ReceiptHandle {receipt_handle}");
 
                 await semaphore.WaitAsync(cancellationToken);
-                _consumerResolver
-                    .Resolve(message, cancellationToken)
-                    .ContinueWith(
-                        task =>
-                        {
-                            if (task.IsFaulted)
-                            {
-                                _logger.LogError(task.Exception,
-                                    "Failed to handle message with id {message_id} and ReceiptHandle {receipt_handle}");
-                            }
-                            else if (task.IsCanceled)
-                            {
-                                _logger.LogWarning(
-                                    "Failed to handle message with id {message_id} and ReceiptHandle {receipt_handle}. Task has been cancelled");
-                            }
-                            else
-                            {
-                                DeleteMessage(queueUrl, message);
-                            }
+                tasks.Add(ProcessEvent(message, cancellationToken, semaphore, queueUrl));
+
+                if (!cancellationToken.IsCancellationRequested)
+                {
+                    continue;
+                }
+                
+                await FinishTasks(tasks);
```

Comment:
You removed parallelism (see defined `MaxNumberOfParallelism` option below) by using async/await. It's not suitable for most of teams.

Reply:
returned previous logic with parallelism